### PR TITLE
Attach tap gesture to custom view

### PR DESCRIPTION
**Pull request by MartinOpde**
Add the possibility to attach the TapGesture to the view you want.
e.g. You want to use it with a drop-down arrow and you want that only tapping the arrow triggers the expand.

### New properties
- ExpandCommandView: the view you want to be recognized as a tap gesture. (Pass it via x:Reference)
- ShouldRotate: bool to indicates if you want rotate animation (default: false)

If ExpandCommandView is not use, it works the same as before.

### CODE REVIEW

File: ExpandableView/ExpandableView.cs [hunk 1]
```diff
@@ -163,11 +179,15 @@ private void OnShouldHandleTapToExpandChanged()
...
 			}
-			PrimaryView.GestureRecognizers.Remove(_defaultTapGesture);
-			if(ShouldHandleTapToExpand)
-			{
-				PrimaryView.GestureRecognizers.Add(_defaultTapGesture);
-			}
+
+            var viewToAttachTapGesture = ExpandCommandView ?? PrimaryView;
+
+            viewToAttachTapGesture.GestureRecognizers.Remove(_defaultTapGesture);
```
Comment (AndreiMisiukevich) [inline ExpandableView/ExpandableView.cs]: We should remove _defaultTapGesture from viewToAttachTapGesture and PrimaryView

**Comment by AndreiMisiukevich** [comment]
Hi) Thanks for your PR, i looked through it.
A few questions please

1) Why don't you want to use ShouldHandleTapToExpand = false and add tapGesture to your Image (arrow), then handle tap and set IsExpanded = true in code behind?

2) ShouldRotate is very specific property, isn't it? What should I do, if i want to rotate only for 90 degree?

**Comment by AndreiMisiukevich** [comment]
Perhaps, it would be more convenient, if we add
1) ExpandAnimationLength property
2) ExpandCollapseStarted event

So,
-> ShouldHandleTapToExpand = false and new tapGesture will allow to set tap handler
And
-> IsExpandChanged and ExpandCollapseStarted will give possibility to rotate arrow

What do you think?

**Comment by AndreiMisiukevich** [comment]
Probably, we can preserve ExpandCommandView... but it should have name "TapHandlerView"..
hm.. as for ShouldRotate I definitely disagree with such approach, because it isn't flexible.

**Comment by MartinOpde** [comment]
Ok, so you will leave the responsibility to animate the arrow (or anything else) to the developer.

For your question 
> Why don't you want to use ShouldHandleTapToExpand = false and add tapGesture to your Image (arrow), then handle tap and set IsExpanded = true in code behind?

I just did not think about it, we can do like that if you prefer.

In the end we keep and rename ExpandCommandView et create IsExpandChanged event?

**Comment by AndreiMisiukevich** [comment]
@MartinOpde I think, it would be awesome =)

Yes, that's right
1) Rename ExpandCommandView to TouchHandlerView
2) viewToAttachTapGesture.GestureRecognizers.Remove(_defaultTapGesture);
viewToAttachTapGesture.GestureRecognizers.Remove(PrimaryView);
3) Add IsExpandChanged event (Should be raised by InvokeAnimation method, i think)
4) Add ExpandAnimationLength property (uint), and pass it to Animation


Could you please apply these changes?

**Comment by MartinOpde** [comment]
I will do it when I have time.

**Comment by MartinOpde** [comment]
@AndreiMisiukevich I made the changes and add a property for the Easing on the expand animation

**Comment by AndreiMisiukevich** [comment]
Very well
Thanks
I will validate it on this week

**Comment by AndreiMisiukevich** [comment]
@MartinOpde available in 1.0.1

**Pull request merged**